Foundation/NSRange.swift

```diff
@@ -99,5 +99,33 @@ public func NSStringFromRange(range: NSRange) -> String {
 }
 
 public func NSRangeFromString(aString: String) -> NSRange {
-    NSUnimplemented()
+    let emptyRange = NSMakeRange(0, 0)
+    if aString.isEmpty {
+        // fail early if the string is empty
+        return emptyRange
+    }
+    let scanner = NSScanner(string: aString)
+    let digitSet = NSCharacterSet.decimalDigitCharacterSet()
+    scanner.scanUpToCharactersFromSet(digitSet)
+    if scanner.atEnd {
+        // fail early if there are no decimal digits
+        return emptyRange
+    }
+    guard let location = scanner.scanInt() else {
```

Comment:
scanInt is perhaps not exactly correct since that would only really be applicable for Int32; NSRange is defined as Int (aka long in c) so the size could technically be up to a 64 bit number on 64 bit platforms; so scanInteger probably is more correct here and also removes the nasty cast later on ;)